modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/CommentParsingTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.sql;
+
+import static org.apache.ignite.internal.sql.engine.util.SqlTestUtils.assertThrowsSqlException;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.ignite.internal.lang.IgniteStringBuilder;
+import org.apache.ignite.internal.logger.IgniteLogger;
+import org.apache.ignite.internal.logger.Loggers;
+import org.apache.ignite.lang.ErrorGroups.Sql;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+/**
+ * Tests to verify parsing of comments.
+ *
+ * <p>Covers:<ol>
+ * <li>E161: SQL comments using leading double minus</li>
+ * </ol>
+ *
+ * <p>According to SQL standard, SQL text containing one or more instances of comment is equivalent to the same SQL text with the comment
+ * replaced with newline.
+ */
+public class CommentParsingTest extends AbstractParserTest {
```

Comment:
let's add test for a query with hints to document how it should works from our perspective

Reply:
yes, sure. What in particular should we check?

Reply:
tests should just document our current behavior - hints don't affect the feature + it will show that we don't use hints for unparsing.

Reply:
hint, in fact, is a another token. Unlike comment, It can't be put in random place. Moreover, all hints _must_ be unparsed

Reply:
seems my thoughts go further for the next feature with bracketed comments